packages/grid/x-data-grid-pro/src/hooks/features/infiniteLoader/useGridInfiniteLoader.ts

```diff
@@ -22,62 +20,48 @@ export const useGridInfiniteLoader = (
   apiRef: React.MutableRefObject<GridPrivateApiPro>,
   props: Pick<
     DataGridProProcessedProps,
-    'onRowsScrollEnd' | 'scrollEndThreshold' | 'pagination' | 'paginationMode' | 'rowsLoadingMode'
+    | 'onRowsScrollEnd'
+    | 'scrollEndThreshold'
+    | 'pagination'
+    | 'paginationMode'
+    | 'rowsLoadingMode'
+    | 'signature'
   >,
 ): void => {
   const visibleColumns = useGridSelector(apiRef, gridVisibleColumnDefinitionsSelector);
   const currentPage = useGridVisibleRows(apiRef, props);
-  const rowsMeta = useGridSelector(apiRef, gridRowsMetaSelector);
-  const contentHeight = Math.max(rowsMeta.currentPageTotalHeight, 1);
+  const observer = React.useRef<IntersectionObserver>();
 
-  const isInScrollBottomArea = React.useRef<boolean>(false);
+  const handleLoadMoreRows = React.useCallback(() => {
+    const viewportPageSize = apiRef.current.getViewportPageSize();
+    const rowScrollEndParam: GridRowScrollEndParams = {
+      visibleColumns,
+      viewportPageSize,
+      visibleRowsCount: currentPage.rows.length,
+    };
 
-  const handleRowsScrollEnd = React.useCallback(
-    (scrollPosition: GridScrollParams) => {
-      const dimensions = apiRef.current.getRootDimensions();
+    apiRef.current.publishEvent('rowsScrollEnd', rowScrollEndParam);
```

Comment:
I played with it a bit and it seems to me that it's being fired to often: https://codesandbox.io/s/infiniteloadinggrid-material-demo-forked-xfdu80?file=/demo.tsx

https://user-images.githubusercontent.com/13808724/236502751-5fcaf797-a698-482d-a76c-f1b484b2596f.mov

1. `onRowsScrollEnd` is called on mount, even though there's no need to fetch additional rows yet
2. `onRowsScrollEnd` is called multiple times on scroll

Is this expected behavior?

Reply:
Ah yes, it seems that there is an initial call that is not needed.
Regarding being called multiple times per scroll - probably debouncing will help here.

I didn't notice it being called when scrolling up?

Reply:
> It's also being called when scrolling up.

Added a check to call the callback only when scrolling down.

Tweaked the solution a but to eliminate unnecessary calls.